src/components/PasswordStrengthCheck/PasswordStrengthCheck.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  DetailedHTMLProps,
+  forwardRef,
+  HTMLAttributes,
+  PropsWithoutRef,
+  useEffect,
+  useState,
+  useCallback,
+} from 'react';
+import { RVColorProp, RVSizeProp, RVVariantProp } from '../../types';
+import { Checkbox } from '../Checkbox';
+import styles from './PasswordStrengthCheck.module.scss';
+
+export interface RVPasswordStrengthCheck
+  extends Omit<
+    PropsWithoutRef<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>,
+    'color' | 'size'
+  > {
+  variant?: Exclude<RVVariantProp, RVVariantProp.disabled>;
+  color?: RVColorProp;
+  size?: RVSizeProp;
+  password?: string;
+  onSuccessfulValidation?: (password: string, validations: Record<string, boolean>) => void;
+  validations: {
+    maxLength?: number | false;
+    minLength?: number | false;
+    withNumberAndSpecialCharacters?: boolean;
+    uppercaseAndLowercase?: boolean;
+  };
+}
+
+const PasswordStrengthCheck = forwardRef<HTMLDivElement, RVPasswordStrengthCheck>(
```

Comment:
Why do you need to have `forwardRef` here?
I think it is better to avoid using it as much as possible.